Document/0x05g-Testing-Network-Communication.md

```diff
@@ -117,6 +117,19 @@ Certificate pinning is the process of associating the backend server with a part
 
 The certificate can be pinned and hardcoded into the app or retrieved at the time the app first connects to the backend. In the latter case, the certificate is associated with ("pinned" to) the host when the host is seen for the first time. This alternative is less secure because attackers intercepting the initial connection can inject their own certificates.
 
+Note that there are various ways to work with a failing pin:
+
+- Inform the user about not being able to connect to the back-end and stop all operations. The app can check whether there is an update and inform the user about updating to the latest version of the app if available.
+- Do a call to a crash-reporting service that the pinning has failed to inform the responsible developer(s) for a security misconfiguration.
+- Only inform the back-end by doing a call with slightly different parameters and no pinning enabled, which automatically informs the back-end that pinning failed. This can either be an HTTP request or a call usling TLS with additional information (e.g. extend an existing JWT token or add another value to a header).
```

Comment:
typo + style guide "back end"
```suggestion
- Only inform the back end by doing a call with slightly different parameters and no pinning enabled, which automatically informs the back end that pinning failed. This can either be an HTTP request or a call using TLS with additional information (e.g. extend an existing JWT token or add another value to a header).
```

Reply:
"with slightly different parameters" -> what are these parameters?

Reply:
See the next sentence. Will restructure